src_nbgl/ui_get_public_key.c

```diff
@@ -2,27 +2,54 @@
 #include "shared_context.h"
 #include "ui_callbacks.h"
 #include "ui_nbgl.h"
+#include "network.h"
 
-static void reviewReject(void) {
+static void cancel_send(void) {
     io_seproxyhal_touch_address_cancel(NULL);
 }
 
-static void confirmTransation(void) {
+static void confirm_send(void) {
     io_seproxyhal_touch_address_ok(NULL);
 }
 
-static void reviewChoice(bool confirm) {
+static void confirm_addr(void) {
+    // display a status page and go back to main
+    nbgl_useCaseStatus("ADDRESS\nVERIFIED", true, confirm_send);
+}
+
+static void reject_addr(void) {
+    nbgl_useCaseStatus("Address verification\ncancelled", false, cancel_send);
+}
+
+static void review_choice(bool confirm) {
     if (confirm) {
-        // display a status page and go back to main
-        nbgl_useCaseStatus("ADDRESS\nVERIFIED", true, confirmTransation);
+        confirm_addr();
     } else {
-        nbgl_useCaseStatus("Address verification\ncancelled", false, reviewReject);
+        reject_addr();
     }
 }
 
-static void buildScreen(void) {
-    nbgl_useCaseAddressConfirmation(strings.common.fullAddress, reviewChoice);
+static void display_addr(void) {
+    nbgl_useCaseAddressConfirmation(strings.common.fullAddress, review_choice);
+}
+
+void ui_display_public_key(const uint64_t *chain_id) {
+    strlcpy(g_stax_shared_buffer, "Verify ", sizeof(g_stax_shared_buffer));
+    if (chain_id != NULL) {
+        if (chain_is_ethereum_compatible(chain_id)) {
+            strlcat(g_stax_shared_buffer,
+                    get_network_name_from_chain_id(chain_id),
+                    sizeof(g_stax_shared_buffer));
+            strlcat(g_stax_shared_buffer, "\n", sizeof(g_stax_shared_buffer));
```

Comment:
What is the longest possible length of `get_network_name_from_chain_id(chain_id),` ?
Is it possible that we end up with
```
"Verify SuperLongNet
work
address"
```
Due to auto wrapping ?

Reply:
If yes, calculate `len(get_network_name_from_chain_id(chain_id))` first and place the `\n` accordingly

Reply:
This should be done by the SDK .....

Reply:
Feels bad... No way to calculate the length and place the '\n' accordingly ? I've done something similar in Exchange when we display the appname of the lib